docs/design/provisioning.md

```diff
@@ -0,0 +1,143 @@
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Abstract
+
+This document proposes a model for the configuration and management of dynamically provisioned persistent volumes in Kubernetes.  Familiarity [Persistent Volumes](../user-guide/persistent-volumes/) is assumed.
+
+Administrators use ConfigMap in a namespace they control to configure many provisioners.  Users request volumes via a selector and/or storage class.
+
+## ConfigMap
+
+Provisioners will be configured using `ConfigMap`.  Configs will be held in a namespace controlled by the administrator.
+
+The admin configures kube-controller-manager with a CLI flag specifying the namespace containing the configs.
+
+`--pv-provisioner-namespace=foobar`
+
+
+### Example Configuration
+
+
+```
+	config := &api.ConfigMap {
+		ObjectMeta: api.ObjectMeta{
+			Name: "gold",
+			Namespace: "foobar",
+			Labels: {
+				"storage-class": "gold",
			}
+		},
+		Data: map[string]string {
+			"plugin-name": "kubernetes.io/gce-pd",
+			"service-account-name": "foo",
+			"params": "${OPAQUE-JSON-MAP}",
		}
+	}
+		
+
+	claim := &api.PersistentVolumeClaim{
+		Spec: api.PersistentVolumeClaimSpec{
+				PersistentVolumeSelector:map[string]string{
+					"storage-class":"gold",
+				},
+			},
+		},
+	}
+	
+```
+
+#### Design Consideration
+
+`StorageClass` has been proposed as a field on PersistentVolumeClaim.  An equivalent of that field is currently stewing as an annotation in the current version of dynamic provisioning.
+
+`PersistentVolumeSelector` on PersistentVolumeClaim has been proposed as a means to select specific volumes, akin to a pod's NodeSelector.
+
+Having both fields is confusing.  Consider this on-premise installation of Kube:
+
+> Admin Joe configures "Gold" to dynamically provision Ceph volumes.  Joe also creates "Silver" volumes by manually provisioning a large number of NFS exports on expensive hardware (SSDs, highly available, stringent backup policy).  Lastly, Joe creates "Bronze" NFS exports on older hardware with slower discs.
+
+> How does a user request Gold/Silver/Bronze?   Using the StorageClass field for "Gold" while requiring the use of VolumeSelector for Silver/Bronze requires a user to know the difference between volumes.
+
+Using just VolumeSelector gives a user a single path to understand.  Admins retain the ability to manually create storage.  Labels on the volume (or the volume creator) are matched with the claim's volume selector.
+
+An additional benefit of using a selector for everything is the ability to deprecate PersistentVolumeAccessModes.  These fields on PV are descriptors only.  They have no functional value other than to describe the capabilities of the volume.  AccessModes on a PVC represent only a way to select certain volumes.  Admins can use labels to describe a volume's capabilities (e.g, ReadWriteMany) and users would select volumes accordingly.
+
+
```

Comment:
I like the idea of depcrecarting the current selection-but-non-enforcement behavior regarding accessModes.